core/src/main/scala/org/http4s/HttpException.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+
+import java.io.IOException
+
+/** Indicates a general failure communicating over HTTP */
+class HttpException(final val message: String = "", final val cause: Option[Throwable] = None)
```

Comment:
This is speculative and maybe should go away.  I'm thinking of problems that happen before we have a request to operate on.

Reply:
Gone.